cornichon-core/src/main/scala/com/github/agourlay/cornichon/http/HttpRequest.scala

```diff
@@ -128,5 +129,16 @@ case class QueryGQL(url: String, query: Document, operationName: Option[String]
     val vars: Map[String, Json] = newVariables.map { case (k, v) ⇒ k → parseJsonUnsafe(v) }(scala.collection.breakOut)
     copy(variables = variables.fold(Some(vars))(v ⇒ Some(v ++ vars)))
   }
+
+  lazy val payload: String = {
+    import io.circe.generic.auto._
+    import io.circe.syntax._
+
+    val querySource = query.source.getOrElse(QueryRenderer.render(query, QueryRenderer.Compact))
```

Comment:
With this change we are now parsing twice the String input:

- once with ```QueryRenderer.Pretty``` for the effect title
- once with ```QueryRenderer.Compact``` for the actual payload sent over the wire

Previously ```QueryRenderer.Pretty``` was used for both. 

Not sure about the real cost of the parsing here, why did you choose this solution?